### PR TITLE
[log-shipper] Bump librdkafka

### PR DESCRIPTION
**Pull request by nabokihms**
Signed-off-by: m.nabokikh <maksim.nabokikh@flant.com>

## Description
* Build librdkafka from the source code.
* Bump rust version.

## Why do we need it, and what problem does it solve?
Fixes https://github.com/deckhouse/deckhouse/issues/3547

## Checklist
- [x] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: log-shipper
type: fix
summary: Bump librdkafka to v2.0.2 to make log-shipper read the full CA certificates chain for Kafka.
impact_level: default
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @nabokihms. I've started the workflow for label 'e2e/run/gcp'...

File: modules/460-log-shipper/images/vector/Dockerfile [hunk 1]
```diff
@@ -1,11 +1,24 @@
...
 RUN apt-get update \
     && apt-get install -yq \
-      ca-certificates make bash cmake libclang1-9 llvm-9 libsasl2-dev librdkafka-dev protobuf-compiler
+      ca-certificates make bash cmake libclang1-9 llvm-9 libsasl2-dev protobuf-compiler software-properties-common
+
+# Install librdkafka-dev >=2.0 because bundled version (1.9.2) has bugs with CA certificates location.
+# https://github.com/confluentinc/librdkafka/commit/f8830a28652532009e3f16854cb9d5004d9de06b
+RUN apt-get update \
+    && apt-get install -yq \
+      wget build-essential libsasl2-dev libcurl4-openssl-dev libzstd-dev libssl-dev zlib1g-dev libc6-dev unzip \
```
Comment (diafour) [inline modules/460-log-shipper/images/vector/Dockerfile:15]:
```suggestion
      wget build-essential libcurl4-openssl-dev libzstd-dev libssl-dev zlib1g-dev libc6-dev unzip \
```

libsasl2-dev is dublicated in previous `RUN`

Comment (nabokihms) [reply modules/460-log-shipper/images/vector/Dockerfile:15]: I did it intentionally to highlight that this is a dependency for both vector and librdkafka. Do you think it is worth deleting?

File: modules/460-log-shipper/images/vector/Dockerfile [hunk 2]
```diff
@@ -20,14 +33,29 @@ RUN cargo build \
...
+    && dpkg -i libssl1.1_1.1.1f-1ubuntu2.16_amd64.deb
+# libcurl.4
+RUN apt-get install -yq libcurl4
+
+COPY --from=build-vector /vector/vector/target/release/vector /usr/bin/vector
+
+COPY --from=build-vector /usr/local/lib/librdkafka.* /usr/local/lib/
+COPY --from=build-vector /usr/local/lib/pkgconfig/rdkafka.pc /usr/local/lib/pkgconfig/rdkafka.pc
+
+ENV LD_LIBRARY_PATH=/usr/local/lib
```
Comment (diafour) [inline modules/460-log-shipper/images/vector/Dockerfile:58]: Seems redundant, `/usr/local/lib` is configured to search in Ubuntu.

Comment (nabokihms) [reply modules/460-log-shipper/images/vector/Dockerfile:58]: Nah, I don't know exactly why, but without this variable, the code does not work. According to the `/etc/ld.so.conf` everything is configured appropriately, but I can see only errors without this env variable.

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @nabokihms. I've started the workflow for label 'e2e/run/gcp'...

File: modules/460-log-shipper/images/vector/Dockerfile (hunk 1 above)
Comment (yalosev) [reply modules/460-log-shipper/images/vector/Dockerfile:15]: I think it's okay. Later we can replace one step with the native librdkafka and the other one will keep working

**Review by yalosev — APPROVED** [review]

**Review by RomanenkoDenys — APPROVED** [review]

**Pull request merged**